src/repository/opamDownload.ml

```diff
@@ -108,13 +110,17 @@ let tool_return url ret =
                 Printf.sprintf "curl: empty response while downloading %s"
                   (OpamUrl.to_string url))
     | l  ->
-      let code = List.hd (List.rev l) in
-      let num = try int_of_string code with Failure _ -> 999 in
-      if num >= 400 then
-        fail (Some ("curl error code " ^ code),
-                  Printf.sprintf "curl: code %s while downloading %s"
-                    code (OpamUrl.to_string url))
-      else Done ()
+      let l = List.rev l in
+      try
+        let code = List.find ((<>) "") l in
+        let num = try int_of_string code with Failure _ -> 999 in
+        if num >= 400 then
+          fail (Some ("curl error code " ^ code),
+                    Printf.sprintf "curl: code %s while downloading %s"
+                      code (OpamUrl.to_string url))
+        else Done ()
```

Comment:
```suggestion
        try
          if int_of_string code >= 400 then
            fail (Some ("curl error code " ^ code),
                  Printf.sprintf "curl: code %s while downloading %s"
                    code (OpamUrl.to_string url))
          else Done ()
        with Failure _ ->
          fail (Some "No error code",
                Printf.sprintf "curl: no error code while downloading %s"
                  (OpamUrl.to_string url))
```
Like that we give the 'no code' information instead of an unknown 999 code

Reply:
Hah - I debated whether the 999 thing should remain as well, so the fact you also thought it suggests it really ought to go!

I pushed an extra commit, but switching it to have fewer `try` blocks (not necessarily elegantly, though). It also removes the separation of "no output" from "no code parsed", which I think is also fine, as we don't really care about the distinction.